src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs

```diff
@@ -118,40 +136,94 @@ private void Process(ComponentIntermediateNode node)
                 //
                 // For example, consider a repeater where the generic type is the 'item' type, but the developer has
                 // not set the items. We won't be able to do type inference on this and so it will just be nonsense.
-                var attributes = node.Attributes.Select(a => a.BoundAttribute).Concat(node.ChildContents.Select(c => c.BoundAttribute));
-                foreach (var attribute in attributes)
+                foreach (var attribute in node.Attributes)
```

Comment:
Here I'm deliberately removing `node.ChildContents` from the set of attributes that are used to resolve generic types at this stage, and moving that to later in the process after we've received any types that cascade from ancestors. This is because `ChildContents` are normally untyped lambdas and hence don't bring any actual type info.

In the case where someone isn't using generic type cascading, then the logic ends up being equivalent to what we had before.